### DISCUSSION TITLE
How to replace certain string in the site at build time?

### DISCUSSION THREAD

**Discussion by seebham** [gatsbyjs/gatsby#38811] [Help]
I have a Gatsby v5 site with Sanity as a CMS. All of my pages are static. Some pages are hardcoded through Gatsby /pages and the remaining are generated by data from Sanity at the build time.

I want to have a feature where I can put a placeholder text like `{current_year}` in the hardcoded pages or in Sanity content blocks and it should be replaced with 2024 (using `new Date().getFullYear().toString()`) at build time.

Can we use [onCreateWebpackConfig](https://www.gatsbyjs.com/docs/reference/config-files/gatsby-node/#onCreateWebpackConfig) or [onCreateBabelConfig](https://www.gatsbyjs.com/docs/reference/config-files/gatsby-node/#onCreateBabelConfig) or using any other lifecycle method?

Grateful for any help/suggestions/lead 🙌🏾

**Comment by 808vita** [comment]
I think you are looking for [page context](https://www.gatsbyjs.com/docs/creating-and-modifying-pages/#pass-context-to-pages) 

Pass context to pages
The automatically created pages can receive context and use that as variables in their GraphQL queries. To override the default and pass your own context, open your site’s gatsby-node.js and add similar to the following:
```

gatsby-node.js

exports.onCreatePage = ({ page, actions }) => {
  const { createPage, deletePage } = actions

  deletePage(page)
  // You can access the variable "house" in your page queries now
  createPage({
    ...page,
    context: {
      ...page.context,
      house: `Gryffindor`,
    },
  })
}
```
On your pages and templates, you can access your context via the prop pageContext like this:

```


import React from "react"

const Page = ({ pageContext }) => {
  return <div>{pageContext.house}</div>
}

export default Page
```
**Page context is serialized before being passed to pages. This means it can’t be used to pass functions into components and Date objects will be serialized into strings.**
